src/mappings/NewEra.ts

```diff
@@ -9,29 +16,94 @@ export async function handleStakersElected(
 }
 
 export async function handleNewEra(event: SubstrateEvent): Promise<void> {
-  const currentEra = (await api.query.staking.currentEra()).unwrap();
+  const currentEra = (await api.query.staking.currentEra()).unwrap().toNumber();
 
-  const exposures = await api.query.staking.erasStakersClipped.entries(
-    currentEra.toBigInt(),
-  );
+  if (api.query.staking.erasStakersOverview) {
+    await processEraStakersPaged(event, currentEra);
+  } else {
+    await processEraStakersClipped(event, currentEra);
+  }
+}
+
+async function processEraStakersClipped(
+  event: SubstrateEvent,
+  currentEra: number,
+): Promise<void> {
+  const exposures =
+    await api.query.staking.erasStakersClipped.entries(currentEra);
 
-  for (const [key, exposure] of exposures) {
+  const eraValidatorInfos = exposures.map(([key, exposure]) => {
     const [, validatorId] = key.args;
+
     let validatorIdString = validatorId.toString();
-    const eraValidatorInfo = EraValidatorInfo.create({
-      id: eventId(event) + validatorIdString,
-      era: currentEra.toNumber(),
-      address: validatorIdString,
-      total: exposure.total.toBigInt(),
-      own: exposure.own.toBigInt(),
-      others: exposure.others.map((other) => {
+    const eraValidatorInfo = new EraValidatorInfo(
+      eventId(event) + validatorIdString,
+      validatorIdString,
+      currentEra,
+      exposure.total.toBigInt(),
+      exposure.own.toBigInt(),
+      exposure.others.map((other) => {
         return {
           who: other.who.toString(),
           value: other.value.toString(),
-        };
+        } as IndividualExposure;
       }),
+    );
+    return eraValidatorInfo.save();
+  });
+
+  await Promise.allSettled(eraValidatorInfos);
+}
+
+async function processEraStakersPaged(
+  event: SubstrateEvent,
+  currentEra: number,
+): Promise<void> {
+  const overview =
+    await api.query.staking.erasStakersOverview.entries(currentEra);
+  const pages = await api.query.staking.erasStakersPaged.entries(currentEra);
+
+  const othersCounted = pages.reduce((accumulator, [key, exp]) => {
```

Comment:
enabled ts strict I found ts reported type error here. 

We add type here ` pages.reduce((accumulator:AccumulatorType, [key, exp]) => {`

``` ts
interface AccumulatorType {
  [key: string]: any; // if you have better type suggestion
}


```